pyhiveapi/apyhiveapi/api/hive_api.py

```diff
@@ -40,7 +41,28 @@ def __init__(self, hiveSession=None, websession=None, token=None):
         self.session = hiveSession
         self.token = token
 
-    def request(self, type, url, jsc=None, camera=False):
+        self.homeID = None
+        if self.session is not None:
+            self.homeID = self.session.config.homeID
+
+    def getParams(self, sendhomeID=False, products=False, devices=False, actions=False):
+        """Get parameters."""
+        params = {
+            "products": products,
+            "devices": devices,
+            "actions": actions,
+        }
+        if sendhomeID and self.homeID is not None:
+            params.update({"homeId": self.homeID})
+        return params
```

Comment:
You need and else if here if someone requests to the send the homeId but its actually set to None. to do this you will also need to create a new exception in the Hive exceptions file and import it into this module

```suggestion
        elif sendhomeID and self.homeID is None:
            raise HiveInvalidHomeId
        return params
```

Reply:
This would change the behaviour of the sync api to require a homeId in the case of one home in hive. I've tried to make sure my change only effects cases where a homeID is actually set.

Reply:
Thinking about it, there's nowhere we don't want to send homeID if it isn't set; except `getHomes`, which doesn't mind it being set. So we don't need an arg for it.

Reply:
What about if your the account owner and you don’t need to send a homeID.

is your thinking to send one anyway?